### PR TITLE
feat: Remove deprecation (MAPCO-4117)

### PR DESCRIPTION
**Pull request by razbroc**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✖                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✔                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✖                                                                        |
| Chore            | ✔                                                                       |

Related issues: #MAPCO-4117

### CODE REVIEW

File: src/clients/jobManagerWrapper.ts [hunk 1]
```diff
@@ -296,6 +170,7 @@ export class JobManagerWrapper extends JobManagerClient {
     }
   }
 
+  //TODO: once will be only one kind of exported jobs, no need to filter by ROI's
```
Comment (asafMasa) [inline src/clients/jobManagerWrapper.ts:170]: please open an issue

File: src/common/interfaces.ts [hunk 2]
```diff
@@ -27,9 +27,6 @@ export interface ICleanupData {
   cleanupExpirationTimeUTC?: Date;
 }
 
-/**
- * @deprecated GetMap API - will be deprecated on future
- */
```
Comment (asafMasa) [inline src/common/interfaces.ts:32]: why was this decorator deleted but the object remained? Do we still need th ICreatePackage? can it be removed?

File: src/common/interfaces.ts [hunk 3]
```diff
@@ -46,13 +43,6 @@ export interface ICallbackBase {
   url: string;
 }
 
-/**
- * @deprecated GetMap API - will be deprecated on future
- */
-export interface ICallbackTarget extends ICallbackBase {
-  bbox: BBox | Polygon;
-}
```
Comment (asafMasa) [inline src/common/interfaces.ts:54]: why do we still need the ICallbackBase? can it be removed?

File: src/common/interfaces.ts [hunk 4]
```diff
@@ -72,17 +62,6 @@ export interface IWorkerInputBase {
   targetFormat?: TileOutputFormat;
 }
 
-/**
- * @deprecated GetMap API - will be deprecated on future
- */
-export interface IWorkerInput extends IWorkerInputBase {
```
Comment (asafMasa) [inline src/common/interfaces.ts:78]: why do we still need the "IWorkerInputBase"? can it be removed?

File: src/common/interfaces.ts [hunk 5]
```diff
@@ -94,37 +73,13 @@ export interface IBasicResponse {
...
-  expirationTime: Date;
-  fileSize: number;
-  dbId: string;
-  packageName: string;
-  targetResolution: number;
-  requestId: string;
-  success: boolean;
-  errorReason?: string;
-}
-
```
Comment (asafMasa) [inline src/common/interfaces.ts:127]: do we still need the "ICallbackDataExportBase"?

File: src/common/interfaces.ts (hunk 2 above)
Comment (asafMasa) [reply src/common/interfaces.ts:32]: Do we still need th IBaseCreatePackage ? can it be removed?

File: src/common/interfaces.ts [hunk 6]
```diff
@@ -194,18 +123,13 @@ export interface JobExportDuplicationParams {
   roi: FeatureCollection;
 }
 
-/**
- * @deprecated GetMap API - will be deprecated on future
- */
```
Comment (asafMasa) [inline src/common/interfaces.ts:199]: why do we still need this object? isn't IJobExportParameters enough?

File: src/common/interfaces.ts [hunk 7]
```diff
@@ -260,14 +184,6 @@ export interface IInput {
   dbId: string;
 }
 
-/**
```
Comment (asafMasa) [inline src/common/interfaces.ts:263]: why do we still need "IInput" interface?

File: src/common/interfaces.ts [hunk 8]
```diff
@@ -304,15 +220,7 @@ export enum ExportVersion {
   ROI = 'ROI',
 }
 
-/**
```
Comment (asafMasa) [inline src/common/interfaces.ts:307]: Please remove the "ExportVersion" enum as we only use the "ROI" scenario

File: src/createPackage/models/createPackageManager.ts [hunk 9]
```diff
@@ -566,90 +368,6 @@ export class CreatePackageManager {
     }
   }
 
-  private generateTileGroups(polygon: Polygon | MultiPolygon, footprint: Polygon | MultiPolygon, zoom: number): ITileRange[] {
```
Comment (asafMasa) [inline src/createPackage/models/createPackageManager.ts:569]: please revert its deletion and put in comment, add:
// TODO: remove and replace with `generateTileGroups` that is commented, when multiple tasks for GPKG target is possible

File: src/finalizationManager.ts [hunk 10]
```diff
@@ -156,34 +156,8 @@ export class FinalizationManager {
...
-        this.logger.info({ jobId: job.id, msg: `GETMAP Execute Failed job finalizing on BBOX (GetMap) exporting for job: ${job.id}` });
-        const gpkgFailedErr = `failed to create gpkg, job: ${job.id}`;
-        await this.taskManager.finalizeJob(job, expirationDateUTC, false, gpkgFailedErr);
-      }
-    }
-    return existsJobs;
+    return roiExistsJobs;
   }
 
   private async handleROIJobs(): Promise<boolean> {
```
Comment (asafMasa) [inline src/finalizationManager.ts]: rename to handleExportJobs

**Review by asafMasa — CHANGES_REQUESTED** [review]

File: src/clients/jobManagerWrapper.ts (hunk 1 above)
Comment (razbroc) [reply src/clients/jobManagerWrapper.ts:170]: https://mapcolonies.atlassian.net/browse/MAPCO-4129

File: src/common/interfaces.ts (hunk 2 above)
Comment (razbroc) [reply src/common/interfaces.ts:32]: removed it all

File: src/common/interfaces.ts (hunk 3 above)
Comment (razbroc) [reply src/common/interfaces.ts:54]: removed

File: src/common/interfaces.ts (hunk 4 above)
Comment (razbroc) [reply src/common/interfaces.ts:78]: removed

File: src/common/interfaces.ts (hunk 5 above)
Comment (razbroc) [reply src/common/interfaces.ts:127]: we still need it. it is used for example in `sendExportCallbacks` and more.

File: src/common/interfaces.ts (hunk 6 above)
Comment (razbroc) [reply src/common/interfaces.ts:199]: removed

File: src/common/interfaces.ts (hunk 7 above)
Comment (razbroc) [reply src/common/interfaces.ts:263]: removed

File: src/common/interfaces.ts (hunk 8 above)
Comment (razbroc) [reply src/common/interfaces.ts:307]: do you want me to keep the enum, for future possible versions? Or remove it and hard code `ROI`/remove the choosing possibility?

File: src/finalizationManager.ts (hunk 10 above)
Comment (razbroc) [reply src/finalizationManager.ts]: fixed

File: src/createPackage/models/createPackageManager.ts (hunk 9 above)
Comment (razbroc) [reply src/createPackage/models/createPackageManager.ts:569]: fixed

File: src/common/interfaces.ts (hunk 5 above)
Comment (asafMasa) [reply src/common/interfaces.ts:127]: please remove the base and leave only the ICallbackExportData as we don't need the base anymore

File: src/common/interfaces.ts (hunk 8 above)
Comment (asafMasa) [reply src/common/interfaces.ts:307]: remove Completely

Comment (razbroc) [reply src/common/interfaces.ts:307]: fixed

File: src/common/interfaces.ts (hunk 5 above)
Comment (razbroc) [reply src/common/interfaces.ts:127]: fixed

File: src/createPackage/models/createPackageManager.ts [hunk 11]
```diff
@@ -3,6 +3,7 @@ import { sep } from 'path';
 import { Logger } from '@map-colonies/js-logger';
 import { Tracer } from '@opentelemetry/api';
 import { withSpanAsyncV4 } from '@map-colonies/telemetry';
+// import { TileRanger } from '@map-colonies/mc-utils';
```
Comment (asafMasa) [inline src/createPackage/models/createPackageManager.ts]: remove

**Review by asafMasa — CHANGES_REQUESTED** [review]

File: src/createPackage/models/createPackageManager.ts (hunk 11 above)
Comment (razbroc) [reply src/createPackage/models/createPackageManager.ts]: fixed

**Review by asafMasa — APPROVED** [review]

**Pull request merged**